streams/shared/src/main/scala/zio/stream/ZChannel.scala

```diff
@@ -1633,23 +1633,15 @@ object ZChannel {
     private[zio] final case class Finalizer[Env, OutErr, OutDone](finalizer: Exit[OutErr, OutDone] => URIO[Env, Any])
         extends Continuation[Env, Any, Any, Any, OutErr, Nothing, Nothing, OutDone, Nothing]
 
-    private[this] def SuccessIdentity(implicit
+    private[zio] def successIdentity[Z](z: Z)(implicit
       trace: Trace
-    ): Any => ZChannel[Any, Any, Any, Any, Nothing, Nothing, Any] =
-      ZChannel.succeedNow(_)
+    ): ZChannel[Any, Any, Any, Any, Nothing, Nothing, Z] =
+      ZChannel.succeedNow(z)
 
-    private[zio] def successIdentity[Z](implicit trace: Trace): Z => ZChannel[Any, Any, Any, Any, Nothing, Nothing, Z] =
```

Comment:
I would be very cautious about removing these. There is a good chance it might be used somewhere or they might be inlined if a library has the inlining enabled and inlines from `zio.**`.

I think it might be better to deprecate them instead and add new ones.

Reply:
Also maybe let's make the new methods private to `stream` instead of `zio`, I don't know why we'd want them private to `zio`

Reply:
@kyri-petrou Both comments were taken into account. Ready for review.

Reply:
This method needs to be deprecated instead of removed